integration-test/2047-restore-early-lake-labels.py

```diff
@@ -0,0 +1,123 @@
+                'kind_detail': 'lake'
+            })
+
+    def test_water_lagoon(self):
+        self.generate_fixtures(dsl.way(1, wkt_loads(
+            'POLYGON ((-122.421611012467 37.80395991667648, '
+            '-122.420227786592 37.80413210674329, -122.420159873957 '
+            '37.80408277780808, -122.420117832801 37.80383748120349, '
+            '-122.42154480663 37.80365748294528, -122.421611012467 '
+            '37.80395991667648))'),
+            {
+            u'natural': u'water',
+            u'way_area': u'6936.45',
+            u'source': u'openstreetmap.org',
+            u'water': u'lagoon',
+            u'name': u'Francisco Reservoir',
+            u'covered': u'no'}))
+
+        self.assert_has_feature(
+            16, 10481, 25324, 'water',
+            {
+                'kind': 'water',
+                'alkaline': True,
+                'kind_detail': 'lake'
+            })
+
+    def test_water_ditch(self):
+        self.generate_fixtures(dsl.way(1, wkt_loads(
+            'POLYGON ((-122.421611012467 37.80395991667648, -122.420227786592 37.80413210674329, -122.420159873957 37.80408277780808, -122.420117832801 37.80383748120349, -122.42154480663 37.80365748294528, -122.421611012467 37.80395991667648))'),
+            {
+            u'natural': u'water',
+            u'way_area': u'6936.45',
+            u'source': u'openstreetmap.org',
+            u'water': u'ditch',
+            u'name': u'Francisco Reservoir',
+            u'covered': u'no'}))
+
+        self.assert_has_feature(
+            16, 10481, 25324, 'water',
+            {
+                'kind': 'water',
+                'kind_detail': 'ditch'
+            })
+
+    def test_name_not_drop(self):
+        import dsl
+
+        lon, lat = (-122.417169, 37.769196)
+
+        for z in range(0, 16):
+            x, y = deg2num(lat, lon, z)
+            area = 265548000
+            self.generate_fixtures(
+                dsl.way(1, dsl.box_area(z, x, y, area), {
+                    'name': u'Clear Lake',
+                    'natural': u'water',
+                    'water': u'lake',
+                    'wikidata': u'Q1099503',
+                    'source': u'openstreetmap.org',
+                }),
+            )
+
+            if z < 6:
+                self.assert_no_matching_feature(z, x, y, 'water',
+                                                {
+                                                    'kind': 'water',
+                                                    'kind_detail': 'lake',
+                                                })
+            if z >= 6:
+                self.assert_has_feature(
+                    z, x, y, 'water', {
+                        'kind': 'water',
+                        'kind_detail': 'lake',
+                        'name': 'Clear Lake',
+                        'min_zoom': 6.0,
+                    })
+
+            # the following assert will start to fail on z == 6
+            if z < 6:
```

Comment:
@nvkelso  starting z==6 there are feature in the water layer


`{'label_placement': True, 'kind': 'water', 'collision_rank': 487, 'name': 'Clear Lake', 'area': 265548000, 'sort_rank': 204, 'kind_detail': 'lake', 'source': 'openstreetmap.org', 'min_zoom': 6.0, 'wikidata_id': 'Q1099503', 'id': 1}`

Reply:
Hmm. For the label we do a remapping of the min_zoom based on area grading of 265548000 sq metershere: https://github.com/tilezen/vector-datasource/blob/master/queries.yaml#L843-L878, so I expect min_zoom 9 for the label.

```
           (8,     500000000),
            (9,     200000000),
            (10,     40000000),
```

Reply:
This looks like an existing bug in that logic, as the last build shows the label for Clear Lake is in fact 6.0 not 9.0 like intended. Let's fix that.

Reply:
The area threshold numbers look fine to me. Is this where clause is malformed?

> (properties is not None and properties.get('label_placement') is True)